SwiftLeeds/Views/Components/CommonTileButtton.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  CommonTileButtton.swift
+//  SwiftLeeds
+//
+//  Created by Alex Logan on 05/07/2022.
+//
+
+import SwiftUI
+
+struct CommonTileButtton<BackgroundType: ShapeStyle>: View {
+    let primaryText: String
+    let secondaryText: String?
+    let primaryColor: Color
+    let secondaryColor: Color
+    var backgroundStyle: BackgroundType
+
+    let onTap: () -> ()
+
+    init(
+        primaryText: String,
+        secondaryText: String?,
+        primaryColor: Color = Color.primary,
+        secondaryColor: Color = Color.secondary,
+        backgroundStyle: Color = Color.cellBackground,
+        onTap: @escaping () -> ()
+    ) where BackgroundType == Color {
+        self.primaryText = primaryText
+        self.secondaryText = secondaryText
+        self.primaryColor = primaryColor
+        self.secondaryColor = secondaryColor
+        self.backgroundStyle = backgroundStyle
+        self.onTap = onTap
+    }
+
+    init(
+        primaryText: String,
+        secondaryText: String?,
+        primaryColor: Color = Color.primary,
+        secondaryColor: Color = Color.secondary,
+        backgroundStyle: BackgroundType,
+        onTap: @escaping () -> ()
+    ) {
+        self.primaryText = primaryText
+        self.secondaryText = secondaryText
+        self.primaryColor = primaryColor
+        self.secondaryColor = secondaryColor
+        self.backgroundStyle = backgroundStyle
+        self.onTap = onTap
+    }
+    
+    var body: some View {
+        Button(action: onTap) {
+            CommonTileView(
+                primaryText: primaryText,
+                secondaryText: secondaryText,
+                primaryColor: primaryColor,
+                secondaryColor: secondaryColor,
+                backgroundStyle: backgroundStyle
+            )
+        }
+        .buttonStyle(SquishyButtonStyle())
```

Comment:
How is the onTap handled via VoiceOver?

Reply:
Buttons work with voiceover for free, so onTap would be fired as expected.

Reply:
Normally I'd use a hint to say what tapping the button does, but didn't realise it adopted the trait for free, nice

Reply:
Its a little scary to trust it at first, but the _generally_ automatic traits are a blessing